docs/docs/api/LayoutAnimations/CustomAnimations.md

```diff
@@ -0,0 +1,196 @@
+---
+id: customAnimations
+title: Custom Animations
+sidebar_label: Custom Animations
+---
+
+If our set of predefined animations is not enough for you then this tab is what you are looking for.
+
+## Custom Exiting Animation
+
+What our exiting animation builders do under the hood is generating a worklet function that returns essential data for starting particular animation. 
+The high level template looks like this:
+
+```js
+function CustomExitingAnimation(startingValues) {
+    'worklet'
+    const animations = {
+        // your animations
+    };
+    const initialValues = {
+        // initial values for animations
+    };
+    return {
+      initialValues,
+      animations,
+    }
+}
+```
+
+* `startingValues` - contains information about where view was displayed and what were its dimentions
+    * `startingValues.originX` - X coordinate of top left corder in parent's coordinate system
+    * `startingValues.originY` - Y coordinate of top left corder in parent's coordinate system
+    * `startingValues.width` - view's width
+    * `startingValues.height` -view's height
+    * `startingValues.globalOriginX` - X coordinate of top left corder in global coordinate system
+    * `startingValues.globalOriginY` - Y coordinate of top left corder in global coordinate system
+
+### Example
+```js
+
+function CardView() {
+  const exiting = (startingValues) => {
+    'worklet';
+    const animations = {
+      originX: withTiming(width, { duration: 3000 }),
+      opacity: withTiming(0.5, { duration: 2000 }),
+    };
+    const initialValues = {
+      originX: startingValues.originX,
+      opacity: 1,
+    };
+  }
+
+
+  return (
+    <Animated.View
+      style={[styles.animatedView]} exiting={exiting} >
+      <Text> Card Example </Text>
+    </Animated.View>
+  );
+}
+
+```
+
+## Custom Entering Animation
+
+What our entering animation builders do under the hood is generating a worklet function that returns essential data for starting particular animation. 
+The high level template looks like this:
+
+```js
+function CustomEnteringAnimation(targetValues) {
+    'worklet'
+    const animations = {
+        // your animations
+    };
+    const initialValues = {
+        // initial values for animations
+    };
+    return {
+      initialValues,
+      animations,
+    }
+}
+```
+
+* `targetValues` - contains information about where view wants to be displayed and what are its dimentions
```

Comment:
What are before values in the context of entering and after values in the context of exiting? 0?

Reply:
There are no before values in exiting/entering animations. There a targetValues (values that we want to reach when the entering animation is finished) and startingValues (values from which we start exiting)

Reply:
But `startingValues` are optional right? Where does it start then?

Reply:
Entering animation only gets `targetValues`. `initialValues` should be obligatory when it comes to Entering Animation. I will add it to docs.